frontend/src/components/ExternalDb.vue

```diff
@@ -53,12 +63,23 @@ export default {
         }, {});
       return orderedRst;
     },
+    dbNameToDisplay() {
+      return this.externalDb.dbName.replace('MetabolicAtlas', 'Metabolic Atlas');
+    },
   },
   async beforeMount() {
-    await this.$store.dispatch('externalDb/getComponentsForExternalDb', {
-      dbName: this.$route.params.dbName,
-      externalId: this.$route.params.identifierId,
-    });
+    try {
+      await this.$store.dispatch('externalDb/getComponentsForExternalDb', {
+        dbName: this.$route.params.dbName,
+        externalId: this.$route.params.identifierId,
+      });
+    } catch {
+      const regex = /^MA[MR]/;
+      const found = this.$route.params.identifierId.match(regex);
```

Comment:
I'm wondering if it wouldn't be more reliable here to use `this.$route.params.dbName`.

Reply:
I added checking for both to make sure the error message is displayed only when querying for MetAtlas IDs and when the ID matches MAR or MAM; fixed in ab38b8ef